lib/web_ui/lib/src/engine/view_embedder/application_dom/custom_element_application_dom.dart

```diff
@@ -0,0 +1,104 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import '../../dom.dart';
+import 'application_dom.dart';
+
+class CustomElementApplicationDom extends ApplicationDom {
+
+  CustomElementApplicationDom(this._hostElement) {
+    // Clear children...
+    while(_hostElement.firstChild != null) {
+      _hostElement.removeChild(_hostElement.lastChild!);
+    }
+
+    // Hook up a resize observer on the hostElement (if supported!).
+    //
+    // Should all this code live in the DimensionsProvider classes?
+    _resizeObserver = createDomResizeObserver(
+      (List<DomResizeObserverEntry> entries, DomResizeObserver _) {
+        entries.forEach(_streamController.add);
+      }
+    );
+
+    assert(() {
+      if (_resizeObserver == null) {
+        domWindow.console.warn('ResizeObserver API not supported. Flutter will not resize with its hostElement.');
+      }
+      return true;
+    }());
+
+    _resizeObserver?.observe(_hostElement);
+  }
+
+  final DomElement _hostElement;
+  late DomResizeObserver? _resizeObserver;
+
+  final StreamController<DomResizeObserverEntry> _streamController =
+    StreamController<DomResizeObserverEntry>.broadcast();
+
+  @override
+  final String type = 'custom-element';
```

Comment:
From the comment above, I don't think we need this as this class can just set the attribute here. However, if we absolutely need it for something, let's call it `debugType` to stress that this is used for debugging only and has no behavioral effect.

Reply:
Removed!